### PR TITLE
Fixes audio streaming issues

### PR DESCRIPTION
**Pull request by aliabid94**
Two issues fixed (linked):
1. Was originally skipping first chunk yielded
2. Streaming really only works with mp3 types so convert wav files to mp3 if streaming.

Fixes: #5168

### CODE REVIEW

File: gradio/components/audio.py [hunk 1]
```diff
@@ -343,11 +343,20 @@ def postprocess(
...
+                sample_rate,
+                format="mp3" if self.streaming else self.format,
             )
             self.temp_files.add(file_path)
         else:
-            file_path = self.make_temp_copy_if_needed(y)
+            if isinstance(y, Path):
+                y = str(y)
+            if self.streaming and not y.endswith(".mp3"):
+                sample_rate, data = processing_utils.audio_from_file(y)
```
Comment (freddyaboulton) [inline gradio/components/audio.py:355]: What's the issue with wav? Do you have an example where it fails? cantina.wav works fine for me without this.

File: gradio/blocks.py [hunk 2]
```diff
@@ -1354,7 +1359,10 @@ def handle_streaming_outputs(
         for i, output_id in enumerate(self.dependencies[fn_index]["outputs"]):
             block = self.blocks[output_id]
             if isinstance(block, StreamableOutput) and block.streaming:
-                stream = block.stream_output(data[i])
+                if end:
+                    stream = None
+                else:
+                    stream = block.stream_output(data[i])
```
Comment (freddyaboulton) [inline gradio/blocks.py:1365]: Why is this needed, isn't the end result of the iterator going to be None anyways?

**Review by freddyaboulton — APPROVED** [review]
Looks good @aliabid94 ! Thanks for the fix. Left some questions but not blocking merge.

File: gradio/components/audio.py (hunk 1 above)
Comment (aliabid94) [reply gradio/components/audio.py:355]: really? cantina.wav specifically does not work for me on chrome. It may be differ browser to browser - the problem with streaming wav's is that wav files have a header in each chunk that can confuse the browser when they are naively chained to gether as a stream - mp3's do not have this problem. As explained to me by chatgpt here: https://chat.openai.com/share/dc4b8dd7-faf3-43f6-98df-ea8d7da23fcc

File: gradio/blocks.py (hunk 2 above)
Comment (aliabid94) [reply gradio/blocks.py:1365]: you're right! will remove

**Comment by abidlabs** [comment]
I'll bump the version and do the release manually -- I won't merge this PR because we should sort out the changeset issue first

**Comment by snarb** [comment]
I don't know gradio very well,  but want to share a general cocern, that reading and writing stream chunks to files microphone/local wav/100 Ms chunks/wav in temp folder/mp3/memory looks problematic for real time streaming to me . Disk Io operations  especially under heavy load are not very predictable and stable from the timing perspective. Something like webrtc looks more promising and easy to maintain as for me

**Comment by freddyaboulton** [comment]
> that reading and writing stream chunks to files

Maybe we can add an option to return bytes directly from your function

**Comment by abidlabs** [comment]
Let's go ahead and merge this in since we've already released this. But I'll leave #5160 open since there's some outstanding issues related to latency
